src/cai_lab_to_nwb/zaki_2024/imaginginterface.py

```diff
@@ -0,0 +1,309 @@
+from roiextractors.imagingextractor import ImagingExtractor
+from roiextractors.multiimagingextractor import MultiImagingExtractor
+from roiextractors.extraction_tools import PathType, DtypeType, get_package
+
+from typing import Optional
+import json
+import datetime
+
+from pydantic import DirectoryPath
+from pathlib import Path
+import numpy as np
+
+from copy import deepcopy
+from pathlib import Path
+from typing import Literal, Optional
+
+import numpy as np
+from pydantic import DirectoryPath, validate_call
+from pynwb import NWBFile
+
+from neuroconv.datainterfaces.ophys.baseimagingextractorinterface import BaseImagingExtractorInterface
+from neuroconv.utils import DeepDict, dict_deep_update
+
+
+class MiniscopeImagingExtractor(MultiImagingExtractor):
+
+    def __init__(self, folder_path: DirectoryPath):
+        self.folder_path = Path(folder_path)
+
+        miniscope_videos_folder_path = self.folder_path / "Miniscope"
```

Comment:
This may not work with Cai lab data. For FC, Neutral Exposure and Recall session the folder containing miniscope data is "miniscope" (all lower case). For Offline sessions the folder is called "My_V4_Miniscope". I think we should pass directly what you called `miniscope_videos_folder_path`

Reply:
Good point.

Reply:
Changed.